src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/haflows/HaFlowPingSpec.groovy

```diff
@@ -0,0 +1,120 @@
+        def swT = topologyHelper.switchTriplets.first()
+        def haFlowRequest = haFlowHelper.randomHaFlow(swT).tap {
+            it.periodicPings = true
+        }
+        def haFlow = haFlowHelper.addHaFlow(haFlowRequest)
+
+        then: "Periodic pings is really enabled"
+        northboundV2.getHaFlow(haFlow.haFlowId).periodicPings
+
+        and: "Packet counter on catch ping rules grows due to pings happening"
+        def sharedSwitchPacketCount = getPacketCountOfVlanPingRule(swT.shared.dpId)
+        def ep1SwitchPacketCount = getPacketCountOfVlanPingRule(swT.ep1.dpId)
+        def ep2SwitchPacketCount = getPacketCountOfVlanPingRule(swT.ep2.dpId)
+
+        Wrappers.wait(pingInterval + WAIT_OFFSET) {
+            def sharedPacketCountNow = getPacketCountOfVlanPingRule(swT.shared.dpId)
+            def ep1PacketCountNow = getPacketCountOfVlanPingRule(swT.ep1.dpId)
+            def ep2PacketCountNow = getPacketCountOfVlanPingRule(swT.ep2.dpId)
+
+            sharedPacketCountNow > sharedSwitchPacketCount && ep1PacketCountNow > ep1SwitchPacketCount &&
+                    ep2PacketCountNow > ep2SwitchPacketCount
+        }
+
+        cleanup:
+        haFlow && haFlowHelper.deleteHaFlow(haFlow.haFlowId)
+    }
+
+    @Tidy
+    @Ignore("unignore after https://github.com/telstra/open-kilda/issues/5224")
+    @Tags([LOW_PRIORITY])
+    def "Able to ping ha-flow when one of subflows is one-switch one"() {
+        given: "ha-flow which has one-switch subflow"
+        def switchTriplet = topologyHelper.getSwitchTriplets(true, true).find {
+            it.shared == it.ep1 && it.shared != it.ep2
+        }
+        assumeTrue(switchTriplet != null, "These cases cannot be covered on given topology:")
+        def haFlow = haFlowHelper.addHaFlow(haFlowHelper.randomHaFlow(switchTriplet))
+
+        and: "expected ping response"
+        def multiSwitchSubFlowId = haFlow.getSubFlows()
+                .find { it.getEndpoint().getSwitchId().equals(switchTriplet.getEp2().getDpId()) }
+                .getFlowId()
+        def expectedResponseSubflowPart = UniSubFlowPingPayload.builder()
+                .pingSuccess(true)
+                .build()
+        def expectedResponse = HaFlowPingResult.builder()
+                .haFlowId(haFlow.getHaFlowId())
+                .pingSuccess(false)
+                .error("One sub flow is one-switch flow")
+                .subFlows([SubFlowPingPayload.builder()
+                                   .flowId(multiSwitchSubFlowId)
+                                   .forward(expectedResponseSubflowPart)
+                                   .reverse(expectedResponseSubflowPart)
+                                   .build()])
+                .build()
+
+        when: "ping ha-flow"
+        def response = northboundV2.pingHaFlow(haFlow.getHaFlowId(), new HaFlowPingPayload(2000))
+        response = 'replace unpredictable latency values from ping response'(response)
```

Comment:
I think it would be better to have `ExpectedHaPingResponse` with builder and overridden `equals()` method. Kind of 
```
given: "Expected ping response"
def expectedPingResponse = ExpectedHaPingResponse.builder().
.haFlowId(haFlow.getHaFlowId())
                .pingSuccess(false)
                .error("One sub flow is one-switch flow")
                .subFlows([SubFlowPingPayload.builder()
                ...
then: "ping response contains expected errors"
expectedPingResponse == haFlowHelper.ping(haFlowId, 2000)
```